src/main/java/md/maib/service/impl/CustomerServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package md.maib.service.impl;
+
+import md.maib.entity.Customer;
+import md.maib.repository.CustomerRepository;
+import md.maib.service.CustomerService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CustomerServiceImpl implements CustomerService {
+
+    private CustomerRepository customerRepository;
+    @Autowired
+    public CustomerServiceImpl(CustomerRepository customerRepository) {
+        this.customerRepository = customerRepository;
+    }
+
+    @Override
+    public Optional<Customer> getCustomerById(Long id) {
+        return customerRepository.findById(id);
+    }
+
+    @Override
+    public List<Customer> getAllCustomers() {
+        return customerRepository.findAll();
+    }
+
+    @Override
+    public Optional<Customer> updateCustomerById(Long id, Customer customerDetails) {
+        return customerRepository.findById(id)
+                .map(customer -> {
```

Comment:
please do not use copilot or other code completion tools at this level. you should write code by yourself, its a typical Junior trap - use ready-made solutions and think that you understand them.